src/System.Private.CoreLib/shared/System/Runtime/CompilerServices/ConditionalWeakTable.cs

```diff
@@ -537,7 +537,7 @@ internal int FindEntry(TKey key, out object? value)
             }
 
             /// <summary>Gets the entry at the specified entry index.</summary>
-            internal bool TryGetEntry(int index, [MaybeNullWhen(false)] out TKey key, [MaybeNullWhen(false)] out TValue value)
+            internal bool TryGetEntry(int index, [MaybeNull, NotNullWhen(true)] out TKey key, [MaybeNullWhen(false)] out TValue value)
```

Comment:
Can we instead use `TKey?` here? Since it is constraint to be of type `class` a `?` should be allowed on `TKey`. It is an internal API, but I want to understand why you chose using `MaybeNull`.

Reply:
Yup, good point.  Will change it.